wgpu-core/src/device/queue.rs

```diff
@@ -1172,7 +1172,9 @@ impl Global {
 
             let mut used_surface_textures = track::TextureUsageScope::default();
 
-            let mut submit_surface_textures_owned = SmallVec::<[_; 2]>::new();
+            // Use a hashmap here to deduplicate the surface textures that are used in the command buffers.
+            // This avoids vulkan deadlocking from the same surface texture being submitted multiple times.
+            let mut submit_surface_textures_owned = FastHashMap::default();
```

Comment:
Should this be an IndexMap instead? I'm not sure if the ordering matters for it.

Reply:
No ordering doesn't matter, just uniqueness.